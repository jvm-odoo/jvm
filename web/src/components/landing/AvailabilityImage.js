import React from 'react'
import styled from 'styled-components'

import me from '../../assets/img/me.jpg'

const AvailabilityImageContainer = styled.div`
  position: relative;
`

const Image = styled.img`
  height: 64px;
  width: 64px;
  border-radius: 50%;
`

const AvailabilityIndicator = styled.span`
  position: absolute;
  bottom: 5px;
  right: 0;
  height: 14px;
  width: 14px;
  background-color: green;
  border: 2px solid #FFF;
  border-radius: 50%;
`

export const AvailabilityImage = () => {
  return (
    <AvailabilityImageContainer>
      <Image src={me} alt="Jason Van Malder" />
      <AvailabilityIndicator />
    </AvailabilityImageContainer>
  )
}