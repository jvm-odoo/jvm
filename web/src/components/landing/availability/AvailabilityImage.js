import React from 'react'
import styled from 'styled-components'

import me from '../../../assets/img/me.jpg'
import { AVAILABILITIES } from '../../../constants'

const AvailabilityImageContainer = styled.div`
  position: relative;
`

const Image = styled.img`
  height: 64px;
  width: 64px;
  border-radius: 50%;
`

const AvailabilityIndicator = styled.span`
  position: absolute;
  bottom: 5px;
  right: 0;
  height: 14px;
  width: 14px;
  background-color: ${({status}) => (
    status === AVAILABILITIES.available ? 'green' :
    status === AVAILABILITIES.partially_available ? 'orange' :
    status === AVAILABILITIES.not_available ? 'red' :
    'grey'
  )};
  border: 2px solid #FFF;
  border-radius: 50%;
`

export const AvailabilityImage = ({ status, loading }) => {
  return (
    <AvailabilityImageContainer>
      <Image src={me} alt="Jason Van Malder" />
      {!loading && (<AvailabilityIndicator status={status} />)}
    </AvailabilityImageContainer>
  )
}