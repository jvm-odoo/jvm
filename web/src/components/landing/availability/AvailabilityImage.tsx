import React from 'react'
import styled, { keyframes } from 'styled-components'

import me from '../../../assets/img/me.jpg'
import { AVAILABILITIES } from '../../../constants'
import { Avatar, Badge, createStyles, withStyles } from '@material-ui/core'
import {IThemeContainer} from '../../../constants/themes'

interface IAvailabilityImage {
    status: number
}

const AvailabilityImageContainer = styled.div`
    position: relative;
`

const Image = styled.img.attrs(() => ({
    loading: 'lazy',
    decoding: 'async',
}))`
    height: 48px;
    width: 48px;
    border-radius: 50%;
    @media only screen and (min-width: 1440px) {
        height: 64px;
        width: 64px;
    }
`

const getColorFromStatus = (status: number) =>
    status === AVAILABILITIES.available
        ? '#6DC36D'
        : status === AVAILABILITIES.partially_available
        ? 'orange'
        : status === AVAILABILITIES.not_available
        ? '#ec5a5a'
        : 'grey'

const availabilityRipple = keyframes`
  from {
    transform: scale(.8);
    opacity: 1;
  }

  to {
    transform: scale(2.4);
    opacity: 0;
  }
`

const AvailabilityIndicator = styled.span<IAvailabilityImage>`
    position: absolute;
    bottom: 5px;
    right: 0;
    height: 10px;
    width: 10px;
    background-color: ${({ status }) => getColorFromStatus(status)};
    border: 2px solid ${({ theme }: IThemeContainer) => theme.background};
    border-radius: 50%;

    &:after {
        position: absolute;
        bottom: -1px;
        right: -1px;
        height: 10px;
        width: 10px;
        border: 1px solid ${({ status }) => getColorFromStatus(status)};
        border-radius: 50%;
        content: ' ';
        animation: ${availabilityRipple} 0.8s infinite ease-in-out;
    }

    @media only screen and (min-width: 1440px) {
        height: 14px;
        width: 14px;

        &:after {
            bottom: 0;
            right: 0;
            height: 12px;
            width: 12px;
        }
    }
`

export const AvailabilityImage = ({ status }: IAvailabilityImage) => {
    return (
        <AvailabilityImageContainer>
            <Image src={me} alt="Jason Van Malder" />
            <AvailabilityIndicator status={status} />
        </AvailabilityImageContainer>
    )
}
