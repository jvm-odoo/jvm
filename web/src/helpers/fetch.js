export const fetchGet = async (url) => {
  const response = await fetch(url)
  return response.json()
}
