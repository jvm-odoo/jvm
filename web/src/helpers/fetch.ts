export const fetchGet = async (url: string) => {
    const response = await fetch(url)
    return response.json()
}

export const fetchPost = async (url: string, data: any, token?: string) => {
    const headers: Headers = new Headers({
        'Content-Type': 'application/json',
        'Access-Control-Allow-Origin': '*'
    })

    if (token) {
        headers.append('Authorization', `Token ${token}`)
    }

    const response = await fetch(url, {
        method: 'POST',
        headers,
        mode: 'cors',
        body: JSON.stringify(data)
    })

    return response.json()
}