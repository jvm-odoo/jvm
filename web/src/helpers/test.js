import React from 'react'
import toJson from 'enzyme-to-json'
import { act } from 'react-dom/test-utils'
import { mount, shallow } from 'enzyme'
import { MockedProvider } from '@apollo/react-testing'

const wait = (time = 0) => new Promise(res => setTimeout(res, time))

export const executeMockProviderTestCase = async (wrapperInstance) => (
  await act(async () => {
    await wait(100)
    wrapperInstance.update()
  })
)

export const shallowMatchSnapshot = (Component) => {
  const wrapper = shallow(Component)
  expect(toJson(wrapper)).toMatchSnapshot()
}

export const mountWrapperWithMockedData = async (Component, mocks) => {
  let wrapper

  act(() => {
    wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        {Component}
      </MockedProvider>
    )
  })

  await executeMockProviderTestCase(wrapper)
  return wrapper
}
