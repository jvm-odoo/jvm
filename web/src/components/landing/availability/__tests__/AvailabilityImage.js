import React from 'react'

import { AvailabilityImage } from '../AvailabilityImage'
import { shallowMatchSnapshot } from '../../../../helpers/test'

it('renders correctly', () => shallowMatchSnapshot(<AvailabilityImage />))
