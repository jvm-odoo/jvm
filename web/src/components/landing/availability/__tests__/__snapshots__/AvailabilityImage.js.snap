// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <styled.img
    alt="Jason Van Malder"
    src="me.jpg"
  />
  <styled.span />
</styled.div>
`;
